DFC.ServiceTaxonomy.PageLocation/Drivers/PageLocationPartDisplayDriver.cs

```diff
@@ -49,15 +50,30 @@ public override async Task<IDisplayResult> UpdateAsync(PageLocationPart model, I
 
         private async Task ValidateAsync(PageLocationPart pageLocation, IUpdateModel updater)
         {
-            if (string.IsNullOrWhiteSpace(pageLocation.UrlName))
+            bool urlNameIsValid = true;
+
+            if (urlNameIsValid && string.IsNullOrWhiteSpace(pageLocation.UrlName))
             {
                 updater.ModelState.AddModelError(Prefix, nameof(pageLocation.UrlName), "A value is required for 'UrlName'");
+                urlNameIsValid = false;
+            }
+
+            if (urlNameIsValid && !Regex.IsMatch(pageLocation.UrlName!, UrlNamePattern))
+            {
+                updater.ModelState.AddModelError(Prefix, nameof(pageLocation.UrlName), $"'UrlName' contains invalid characters. Valid characters include A-Z, 0-9, '-' and '_'.");
+                urlNameIsValid = false;
+            }
+
+            if (urlNameIsValid && pageLocation.UrlName!.All(c => !char.IsLetterOrDigit(c)))
+            {
+                updater.ModelState.AddModelError(Prefix, nameof(pageLocation.UrlName), $"'UrlName' must contain at least one alphanumeric character.");
+                urlNameIsValid = false;
             }
 
-            if (pageLocation.UrlName?.IndexOfAny(InvalidCharactersForPath) > -1 || pageLocation.UrlName?.IndexOf(' ') > -1)
+            if (urlNameIsValid && pageLocation.UrlName!.All(c => !char.IsLetter(c)))
```

Comment:
As this is more restrictive than the previous check, can't we just have this check?